arch/x86/core/pcie.c

```diff
@@ -206,38 +206,19 @@ uint32_t pcie_msi_map(unsigned int irq,
 uint16_t pcie_msi_mdr(unsigned int irq,
 		      msi_vector_t *vector)
 {
-#ifdef CONFIG_PCIE_MSI_X
-	if ((vector != NULL) && (vector->msix)) {
-		return 0x4000U | vector->arch.vector;
-	}
-#endif
+	uint16_t msg_data_reg = 0;
 
-	if (vector == NULL) {
+	if (vector != NULL) {
+		msg_data_reg = vector->arch.vector;
+	} else {
 		/* edge triggered */
-		return 0x4000U | Z_IRQ_TO_INTERRUPT_VECTOR(irq);
+		msg_data_reg = 0x4000U | Z_IRQ_TO_INTERRUPT_VECTOR(irq);
```

Comment:
that is not going to work if there is no default irq set for the msi device, see below

Reply:
This will not have any impact on removing code from line number 273 to 278, since that code will come into picture only for multi vector (n_vector > 1). But any way will put back that code with TODO comment to fix the design